### PR TITLE
update DfmIncludeBlockRule.

### PR DESCRIPTION
**Pull request by vwxyzh**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Dfm/Rules/DfmIncludeBlockRule.cs [hunk 1]
```diff
@@ -7,15 +7,78 @@ namespace Microsoft.DocAsCode.Dfm
     using System.Text.RegularExpressions;
 
     using Microsoft.DocAsCode.MarkdownLite;
+    using Microsoft.DocAsCode.MarkdownLite.Matchers;
 
     public class DfmIncludeBlockRule : IMarkdownRule
     {
+        private static readonly Matcher _IncludeMatcher =
+            Matcher.WhiteSpacesOrEmpty +
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Dfm/Rules/DfmIncludeBlockRule.cs]: >WhiteSpacesOrEmpty [](start = 20, length = 18)

from previous regex, it should not start with spaces?

File: src/Microsoft.DocAsCode.Dfm/Rules/DfmIncludeBlockRule.cs [hunk 2]
```diff
@@ -7,15 +7,78 @@ namespace Microsoft.DocAsCode.Dfm
...
+            {
+                var sourceInfo = context.Consume(match.Length);
+
+                // [!include[title](path "optionalTitle")]
+                // 1. Get include file path 
+                var path = match["path"].GetValue();
+
+                // 2. Get title
+                var value = match["title"].GetValue();
+                var title = match.GetGroup("optionalTitle")?.GetValue() ?? string.Empty;
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Dfm/Rules/DfmIncludeBlockRule.cs]: >title [](start = 20, length = 5)

minor: rename to optionalTitle? seems above var should be title?

File: src/Microsoft.DocAsCode.Dfm/Rules/DfmIncludeBlockRule.cs [hunk 3]
```diff
@@ -7,15 +7,78 @@ namespace Microsoft.DocAsCode.Dfm
...
     {
+        private static readonly Matcher _IncludeMatcher =
+            Matcher.WhiteSpacesOrEmpty +
+            "[!" +
+            Matcher.CaseInsensitiveString("include") +
+            Matcher.Char('+').Maybe() +
+            Matcher.WhiteSpacesOrEmpty +
+            '[' +
+            (
+                Matcher.AnyCharNot(']').RepeatAtLeast(1) |
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Dfm/Rules/DfmIncludeBlockRule.cs]: >RepeatAtLeast(1 [](start = 40, length = 15)

RepeatAtLeast(0)?

**Comment by superyyrrzz** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Dfm/Rules/DfmIncludeBlockRule.cs (hunk 1 above)
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Dfm/Rules/DfmIncludeBlockRule.cs]: Not same with regex, but I think it should be allowed.

---
In reply to: [107637464](https://github.com/dotnet/docfx/pull/1472#discussion_r107637464) [](ancestors = 107637464)

File: src/Microsoft.DocAsCode.Dfm/Rules/DfmIncludeBlockRule.cs (hunk 3 above)
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Dfm/Rules/DfmIncludeBlockRule.cs]: Yes, `.RepeatAtLeast(0).ToGroup("title")`

---
In reply to: [107638237](https://github.com/dotnet/docfx/pull/1472#discussion_r107638237) [](ancestors = 107638237)

**Comment by hellosnow** [comment]
:shipit:

**Pull request merged**